src/safeposix/filesystem.rs

```diff
@@ -70,21 +70,53 @@ pub struct FilesystemMetadata {
     pub inodetable: interface::RustHashMap<usize, Inode>,
 }
 
+pub fn init_filename_to_inode_dict(parentinode: usize) -> interface::RustHashMap<std::ffi::OsString, usize> {
+    let mut retval = interface::RustHashMap::new();
+    retval.insert(std::ffi::OsString::from("."), ROOTDIRECTORYINODE);
+    retval.insert(std::ffi::OsString::from(".."), parentinode);
+    retval
+}
+
 impl FilesystemMetadata {
     pub fn blank_fs_init() -> FilesystemMetadata {
         //remove open files?
         let mut retval = FilesystemMetadata {nextinode: STREAMINODE + 1, dev_id: 20, inodetable: interface::RustHashMap::new()};
         let time = interface::timestamp(); //We do a real timestamp now
-        let mut dirin = DirectoryInode {size: 0, uid: DEFAULT_UID, gid: DEFAULT_GID, 
-            mode: S_IFDIR as u32 | S_IRWXA, atime: time, ctime: time, mtime: time,
-            linkcount: 3, refcount: 0, filename_to_inode_dict: interface::RustHashMap::new()};//this is where cwd starts
-        dirin.filename_to_inode_dict.insert(std::ffi::OsString::from("."), ROOTDIRECTORYINODE);
-        dirin.filename_to_inode_dict.insert(std::ffi::OsString::from(".."), ROOTDIRECTORYINODE);
-        retval.inodetable.insert(ROOTDIRECTORYINODE, Inode::Dir(dirin));
+        let dirinode = DirectoryInode {size: 0, uid: DEFAULT_UID, gid: DEFAULT_GID,
+            mode: S_IFDIR as u32 | S_IRWXA, linkcount: 3, refcount: 0, 
+            atime: time, ctime: time, mtime: time,
+            filename_to_inode_dict: init_filename_to_inode_dict(ROOTDIRECTORYINODE)};
+        retval.inodetable.insert(ROOTDIRECTORYINODE, Inode::Dir(dirinode));
         retval
     }
 }
 
+pub fn load_fs_special_files() {
```

Comment:
Is this where we're setting up the initial cage now?

Reply:
No I haven't really settled on how/if the initial cage is going to work

Reply:
OK this is fine for now, we'll probably move this around once I do my PR.